src/StakingNode.sol

```diff
@@ -472,46 +531,17 @@ contract StakingNode is IStakingNode, StakingNodeEvents, ReentrancyGuardUpgradea
         return delegatedTo == delegationManager.delegatedTo(address(this));
     }
 
-    function synchronize(uint256 queuedShares, uint32 undelegateBlockNumber) public onlyDelegator {
-        if (isSynchronized()) { 
-            revert AlreadySynchronized();
-        }
-
-        IDelegationManager delegationManager = IDelegationManager(address(stakingNodesManager.delegationManager()));
-
-        IStrategy[] memory strategies = new IStrategy[](1);
-        uint256[] memory shares = new uint256[](1);
-        strategies[0] = beaconChainETHStrategy;
-        shares[0] = queuedShares;
-
-        address thisNode = address(this);
-
-        // We can assume that the Withdrawal queued by the undelegate() call made by the operator
-        // is the LAST queued withdrawal since to call queueWithdrawals you require isSynchronized() == true.
-        IDelegationManagerTypes.Withdrawal memory withdrawal = IDelegationManagerTypes.Withdrawal({
-            staker: thisNode,
-            delegatedTo: delegatedTo,
-            withdrawer: thisNode,
-            nonce: delegationManager.cumulativeWithdrawalsQueued(thisNode) - 1, // must be the last withdrawal
-            startBlock: undelegateBlockNumber,
-            strategies: strategies,
-            scaledShares: shares
-        });
-
-        // IMPORTANT: withdrawalRoot is not spoofable because nonce is a strictly increasing value that
-        // gets incremented for each new withdrawal.
-        // This function only works with this pre-condition
-        bytes32 withdrawalRoot = delegationManager.calculateWithdrawalRoot(withdrawal);
-
-        // Withdrawal MUST exist and must be the last
-        if (!IDelegationManagerExtended(address(delegationManager)).pendingWithdrawals(withdrawalRoot)) {
-            revert WithdrawalMismatch();
-        }
+    /**
+     * @notice Synchronizes the StakingNode's delegation state with the DelegationManager and queued shares.
+     * @dev This function should be called after operator undelegate to this StakingNode or there is slashing event.
+     */
+    function synchronize() public onlyDelegator {
 
-        // queue shares
-        queuedSharesAmount += queuedShares;
+        syncQueuedShares();
 
-        delegatedTo = address(0);
+        IDelegationManager delegationManager = IDelegationManager(address(stakingNodesManager.delegationManager()));
+        delegatedTo = delegationManager.delegatedTo(address(this));
+        stakingNodesManager.updateTotalETHStaked();
```

Comment:
I like this idea of having the staking node, calling the update on the manager.

For the token staking node we thought of adding a function that calls synchronize on all nodes and the updates balances.

But you solution seems better given that only a particular operator is slashed at a time and synchronizing all nodes at the same time seems unnecessary.